example-restapi-rest-assured/src/main/java/com/estafet/qapractices/core/ReqRest.java

```diff
@@ -0,0 +1,182 @@
+package com.estafet.qapractices.core;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.assertj.core.api.SoftAssertions;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+
+import com.estafet.qapractices.api.ApiHooks;
+import com.estafet.qapractices.context.Context;
+import com.estafet.qapractices.model.ReqResUserModel;
+import com.estafet.qapractices.model.ReqResUsersModel;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import io.restassured.response.ValidatableResponse;
+
+/* Created by Anton Rusanov on 28/05/18
+ *  This class contains methods for basic requests with Rest Assured.
+ */
+public class ReqRest {
+
+	private Context context;
+	
+	private final String GETPATH = "/users/";
+	private final String POSTPATH = "/users/";
+	private final String PUTPATH = "/users/";
+	private final String DELETEPATH = "/users/";
+	private Response response;
+	private ValidatableResponse validateResponse;
+	private ApiHooks apiHooks;
+
+	@Inject
+	public ReqRest(Context context) {
+		this.context = context; 
+		
+		
+		/*
+		 * baseURI method will set globally the URI that you want to connect. It have to
+		 * be setted into a constructor or method. After that in given() method of
+		 * RestAssured class, this URI will be loaded automatically.
+		 */
+		RestAssured.baseURI = "https://reqres.in/api";
```

Comment:
the hooks doesn't work for now, maybe is because we have injection. I'm still in research for this problem. And yes if hooks works, of course we don't have to set up the URI twice.